src/main/java/org/apache/skywalking/banyandb/v1/client/MeasureBulkWriteProcessor.java

```diff
@@ -33,30 +39,50 @@
 @ThreadSafe
 public class MeasureBulkWriteProcessor extends AbstractBulkWriteProcessor<BanyandbMeasure.WriteRequest,
         MeasureServiceGrpc.MeasureServiceStub> {
+    private final BanyanDBClient client;
+
     /**
      * Create the processor.
      *
-     * @param measureServiceStub stub for gRPC call.
+     * @param client             the client
      * @param maxBulkSize        the max bulk size for the flush operation
      * @param flushInterval      if given maxBulkSize is not reached in this period, the flush would be trigger
      *                           automatically. Unit is second.
      * @param concurrency        the number of concurrency would run for the flush max.
      */
     protected MeasureBulkWriteProcessor(
-            final MeasureServiceGrpc.MeasureServiceStub measureServiceStub,
+            final BanyanDBClient client,
             final int maxBulkSize,
             final int flushInterval,
             final int concurrency) {
-        super(measureServiceStub, "MeasureBulkWriteProcessor", maxBulkSize, flushInterval, concurrency);
+        super(client.getMeasureServiceStub(), "MeasureBulkWriteProcessor", maxBulkSize, flushInterval, concurrency);
+        this.client = client;
     }
 
     @Override
     protected StreamObserver<BanyandbMeasure.WriteRequest> buildStreamObserver(MeasureServiceGrpc.MeasureServiceStub stub,
                                                                                CompletableFuture<Void> batch) {
         return stub.write(new StreamObserver<BanyandbMeasure.WriteResponse>() {
+            private final Set<String> schemaExpired = new HashSet<>();
+
             @Override
             public void onNext(BanyandbMeasure.WriteResponse writeResponse) {
-
+                switch (writeResponse.getStatus()) {
+                case STATUS_EXPIRED_REVISION:
```

Comment:
I prefer to handle them. BanyanDB batch writing services log these errors instead of returning them to the clients. In this round, I would like the Java client to return these errors through `CompletableFuture`. OAP module will have a chance to log them.

Reply:
The server unexpected errors will override these errors, which is reasonable.